addons/sourcemod/scripting/ConnectAnnounce.sp

```diff
@@ -1062,118 +1070,104 @@ public void Announcer(int client, int iRank, bool sendToAll)
-		}
-		else
-		{
-			ReplaceString(sFinalMessage, sizeof(sFinalMessage), "{NOSTEAM}", "");
-		}
+		char sBuffer[16];
+		if (g_bNative_Connect && !SteamClientAuthenticated(g_sAuthID[client]))
+			Format(sBuffer, sizeof(sBuffer), " <NoSteam>");
+
+		ReplaceString(sFinalMessage, sizeof(sFinalMessage), "{NOSTEAM}", sBuffer);
 	}
 #endif
 
 #if defined _EntWatch_include
 	if (StrContains(sFinalMessage, "{EBANS}"))
 	{
+		char sBuffer[16];
 		if (g_bNative_EntWatch)
 		{
-			char sBuffer[16];
 			int iEntWatch = EntWatch_GetClientEbansNumber(client);
-			Format(sBuffer, sizeof(sBuffer), "%d", iEntWatch);
-			ReplaceString(sFinalMessage, sizeof(sFinalMessage), "{EBANS}", sBuffer);
-		}
-		else
-		{
-			ReplaceString(sFinalMessage, sizeof(sFinalMessage), "{EBANS}", "");
+			FormatBanCount(sBuffer, sizeof(sBuffer), iEntWatch, "EBans");
 		}
+
+		ReplaceString(sFinalMessage, sizeof(sFinalMessage), "{EBANS}", sBuffer);
 	}
 #endif
 
 #if defined _KnockbackRestrict_included_
 	if (StrContains(sFinalMessage, "{KBANS}"))
 	{
+		char sBuffer[16];
 		if (g_bNative_KbRestrict)
 		{
-			char sBuffer[16];
 			int iKbans = KR_GetClientKbansNumber(client);
-			Format(sBuffer, sizeof(sBuffer), "%d", iKbans);
-			ReplaceString(sFinalMessage, sizeof(sFinalMessage), "{KBANS}", sBuffer);
-		}
-		else
-		{
-			ReplaceString(sFinalMessage, sizeof(sFinalMessage), "{KBANS}", "");
+			FormatBanCount(sBuffer, sizeof(sBuffer), iKbans, "KBans");
 		}
+
+		ReplaceString(sFinalMessage, sizeof(sFinalMessage), "{KBANS}", sBuffer);
 	}
 #endif
 
 #if defined _sourcebanschecker_included
 	if (StrContains(sFinalMessage, "{BANS}"))
 	{
+		char sBuffer[16];
 		if (g_bNative_SbChecker_Bans)
 		{
-			char sBuffer[16];
 			int iSbans = SBPP_CheckerGetClientsBans(client);
-			Format(sBuffer, sizeof(sBuffer), "%d", iSbans);
-			ReplaceString(sFinalMessage, sizeof(sFinalMessage), "{BANS}", sBuffer);
-		}
-		else
-		{
-			ReplaceString(sFinalMessage, sizeof(sFinalMessage), "{BANS}", "");
+			FormatBanCount(sBuffer, sizeof(sBuffer), iSbans, "Bans");
 		}
+
+		ReplaceString(sFinalMessage, sizeof(sFinalMessage), "{BANS}", sBuffer);
 	}
 
+	// Todo: When sbpp checker will support mute and gag separately add {MUTES} and {GAGS}
+
 	if (StrContains(sFinalMessage, "{COMMS}"))
 	{
+		char sBuffer[16];
 		if (g_bNative_SbChecker_Comms)
 		{
-			char sBuffer[16];
 			int iComms = SBPP_CheckerGetClientsComms(client);
-			Format(sBuffer, sizeof(sBuffer), "%d", iComms);
-			ReplaceString(sFinalMessage, sizeof(sFinalMessage), "{COMMS}", sBuffer);
-		}
-		else
-		{
-			ReplaceString(sFinalMessage, sizeof(sFinalMessage), "{COMMS}", "");
+			FormatBanCount(sBuffer, sizeof(sBuffer), iComms, "Comms");
 		}
+
+		ReplaceString(sFinalMessage, sizeof(sFinalMessage), "{COMMS}", sBuffer);
 	}
 #endif
 
 	if (StrContains(sFinalMessage, "{STEAMID}"))
 	{
-		ReplaceString(sFinalMessage, sizeof(sFinalMessage), "{STEAMID}", g_sAuthID[client]);
+		char sBuffer[32];
+		switch (g_hCvar_AuthIdType.IntValue)
```

Comment:
You can directly call the fonction with a static cast of this enum:
- https://github.com/alliedmodders/sourcemod/blob/3bc55e68900459103b59f387a5106a87956c7a41/plugins/include/clients.inc#L57

But make sure that the cvar respects the order defined by the enum

Reply:
Yes, but I personnaly think it's become more complicated for the cvar, since steamid2 become cvar = 1, Steamid3 becone cvar 2, and Steamid64 become 3.
What you think ?

Reply:
This order seems logic to me since it follows a convention of SteamX with X going from 0 to Y